Add $(FPIC) build flags to fix LTO builds.

Jalakas opened the pull request:
Add $(FPIC) build flags to fix LTO builds.

hauke commented:
In most builds `$(FPIC)` should be set, we activate it here if `CONFIG_PKG_ASLR_PIE_REGULAR` is set:
https://github.com/openwrt/openwrt/blob/main/include/hardening.mk#L25

Are you building without `CONFIG_PKG_ASLR_PIE_REGULAR`?

Jalakas commented:
> Are you building without `CONFIG_PKG_ASLR_PIE_REGULAR`?
Yes.

> In most builds `$(FPIC)` should be set, we activate it here if `CONFIG_PKG_ASLR_PIE_REGULAR` is set: https://github.com/openwrt/openwrt/blob/main/include/hardening.mk#L25

As much as I understand the chain is:
1. https://github.com/openwrt/openwrt/blob/main/include/hardening.mk#L23 checks if CONFIG_PKG_ASLR_PIE_REGULAR is set, then on line
2. https://github.com/openwrt/openwrt/blob/main/include/hardening.mk#L24 it is checked if PKG_ASLR_PIE_REGULAR is set, but line
3. https://github.com/openwrt/openwrt/blob/main/include/hardening.mk#L7 has set PKG_ASLR_PIE_REGULAR to **0** if it is not set on makefile

Jalakas committed:
build: Add !USE_APK dependency to CLEAN_IPKG to prevent build errors
When USE_APK is enabled, APK replaces opkg/ipkg, leading to a build failure when CLEAN_IPKG is also selected.
Add 'depends on !USE_APK' to CLEAN_IPKG to ensure this option is only selectable when opkg/ipkg is the active package manager.

Signed-off-by: Anari Jalakas <anari.jalakas@gmail.com>
